src/ceph-daemon/ceph-daemon

```diff
@@ -1305,9 +1305,12 @@ def list_daemons():
                     (enabled, state) = check_unit('ceph-%s@%s' % (daemon_type,
                                                                   daemon_id))
                     if not host_version:
-                        out, err, code = call(['ceph', '-v'])
-                        if not code and out.startswith('ceph version '):
-                            host_version = out.split(' ')[2]
+                        try:
+                            out, err, code = call(['ceph', '-v'])
+                            if not code and out.startswith('ceph version '):
+                                host_version = out.split(' ')[2]
```

Comment:
That looks so brittle to me. Anyway. Does it make sense to have a dedicated function for it?

Reply:
Probably.. there are 2 instance right now.  Although I'm trying to change the container case to get this from a label on the container image instead (faster!).  See https://github.com/ceph/ceph-container/issues/1508

Reply:
i'll clean it up after the label thing

Reply:
ceph/ceph-container#1508 will make this block of code much cleaner  :+1: